core/trie/trie.go

```diff
@@ -164,8 +171,13 @@ func (t *Trie) Get(key *felt.Felt) (*felt.Felt, error) {
 }
 
 // Put updates the corresponding `value` for a `key`
+//
+//nolint:gocyclo
 func (t *Trie) Put(key, value *felt.Felt) (*felt.Felt, error) {
-	// Todo: check key is not bigger than max key value for a trie height.
+	if key.Cmp(t.maxKey) > 0 {
```

Comment:
I expect this to have negative effects on performance, but hoping it to be negligible.
`Cmp` implementation is bunch of `if` statements, that many conditionals are bound to mess up branch prediction.

Reply:
I believe modern compilers try to account for this and performance impact is minimal.